backend/tournesol/views/mixins/poll.py

```diff
@@ -0,0 +1,46 @@
+from django.core.exceptions import ObjectDoesNotExist
+from rest_framework import status
+from rest_framework.response import Response
+
+from tournesol.models import Poll
+
+
+class PollScopedViewMixin:
+    """
+    A mixin view that automatically retrieves the poll name from a URL path
+    parameter, and make it available in the view and the serializer context.
+
+    This view must be mixed with a `rest_framework.generics.GenericAPIView`
+    view or a subclass of it.
+    """
+    poll_parameter = "poll_name"
+    # used to avoid multiple similar database queries in a single HTTP request
+    poll_from_url: Poll
+
+    def poll_from_kwargs_or_404(self, request_kwargs):
+        try:
+            return Poll.objects.get(name=request_kwargs[self.poll_parameter])
+        except ObjectDoesNotExist:
+            return self.response_404_poll_doesnt_exist(request_kwargs[self.poll_parameter])
+
+    def response_404_poll_doesnt_exist(self, poll_name: str):
+        return Response(
+            {
+                "detail": "The requested poll {0} doesn't exist.".format(poll_name),
+            },
+            status=status.HTTP_404_NOT_FOUND,
+        )
```

Comment:
Returning a 404 response is not correct here, as `.initial()` is unable to handle it.

An exception should be used instead, possibly with `get_object_or_404` shortcut provided by Django:

```suggestion
    def poll_from_kwargs_or_404(self, request_kwargs):
        return get_object_or_404(Poll, name=request_kwargs[self.poll_parameter])
```